microsoft-edge/devtools-guide-chromium/performance/extension.md

```diff
@@ -0,0 +1,522 @@
+
+* `console.timeStamp` API:
+
+   * Performance impact of instrumentation is a primary concern, especially in production builds.
+
+   * You need a quick and efficient way to mark durations or events without the need for extra metadata.
+
+   * You only need the data to be visualized within the **Performance** tool.
+
+* **User Timings API (performance.mark, performance.measure):**
+
+   * Use when you need to store extra data with each entry, and when you are already using the User Timings API.
+
+   * You need to associate rich data (tooltips, detailed properties) with your performance entries.
+
+   * You want to add visual markers to highlight specific moments.
+
+   * You require the data to be available not only in DevTools but also in the browser's internal performance timeline for broader analysis or other tools.
+
+   * You are already familiar with or using the User Timings API.
+
+
+<!-- ====================================================================== -->
+## Inject your data with `console.timeStamp`
+<!-- https://developer.chrome.com/docs/devtools/performance/extension#inject_your_data_with_consoletimestamp -->
+
+The `console.timeStamp` API is extended to allow for the creation of custom timing entries in the **Performance** tool with minimal overhead, especially when DevTools is not recording a trace.
+
+Syntax:
+
+```javascript
+console.timeStamp(label: string, 
+                  start?: string|number, 
+                  end?: string|number, 
+                  trackName?: string, 
+                  trackGroup?: string, 
+                  color?: DevToolsColor);
+```
+
+* `label`:
+
+   The label for the timing entry.
+
+* `start` (optional):
+
+   * If defined as string: the name of a previously recorded timestamp (using `console.timeStamp(timeStampName)`).
+
+   * If defined as number: a timestamp in milliseconds relative to [Performance.timeOrigin](https://developer.mozilla.org/docs/Web/API/Performance/timeOrigin) (for example, taken with [performance.now()](https://developer.mozilla.org/docs/Web/API/Performance/now)) that represents the start time of the timing entry.
+
+   * If undefined, the current time is used as start time.
+
+* `end`:
+
+   * If defined as string: The name of a previously recorded timestamp.
+
+   * If defined as number: a timestamp in milliseconds relative to [Performance.timeOrigin](https://developer.mozilla.org/docs/Web/API/Performance/timeOrigin) (for example, taken with [performance.now()](https://developer.mozilla.org/docs/Web/API/Performance/now)) that represents the end time of the timing entry.
+
+   * If undefined, the current time is used as end time.
+
+* `trackName`:
+
+   * The name of the custom track.
+
+* `trackGroup`:
+
+   * The name of the track group.
+
+* `color`:
+
+   * The color of the entry.
+
+See also:
+* [console: timeStamp() static method](https://developer.mozilla.org/docs/Web/API/console/timeStamp_static)<!-- todo: better link? only lists 1 param, not 6 -->
```

Comment:
Is there a `console.timeStamp()` Ref link that lists 6 params (custom for DevTools) instead of just 1 param?
https://developer.mozilla.org/docs/Web/API/console/timeStamp_static
@captainbrosset